config/import-control.xml

```diff
@@ -109,20 +119,20 @@
   </subpackage>
 
   <subpackage name="doclets">
+    <allow class="java.nio.charset.StandardCharsets" local-only="true"/>
     <allow pkg="com.sun.javadoc"/>
     <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.(checks|ant|filters|gui)" regex="true"/>
   </subpackage>
 
   <subpackage name="filters">
-    <allow pkg="java.lang.ref"/>
-    <!-- check's subpackages -->
+    <allow class="java.lang.ref.WeakReference" local-only="true"/>
     <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.checks\.[^.]+" regex="true"/>
-    <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.(ant|doclets|gui)" regex="true"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle.utils"/>
   </subpackage>
 
   <subpackage name="gui">
     <allow pkg="java.awt"/>
     <allow pkg="javax.swing"/>
-    <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.(checks|ant|doclets|filters)" regex="true"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle"/>
```

Comment:
it sounds like alot ... can we restrict usage of other packages as much as possible ?